### ISSUE TITLE
Servers: allow bot owner to specify a role to assign new users

### ISSUE THREAD

**Issue by xNifty** [snoringninja/niftybot-discord#35]
Allow server owners to set a role that all new users joining the discord should be assigned to.  This should have a toggle for enabled/disabled, and should fire when on_member_join is triggered.

**Comment by netwrksolutions** [comment]
Could it be possible to also add a timer so it can either auto assign or assign after a selected time?  For instance, when the pug raiders join our server the Lead running the raid gives them @raiders, new members get a different rank.  I'd like the Lead to be able to assign the raiders and if after say 5 minutes if they are not assigned we auto assign.

**Comment by xNifty** [comment]
[**@netwrksolutions**](https://github.com/netwrksolutions) commented on [Dec 1, 2017, 5:59 PM EST](https://github.com/xNifty/niftybot-discord/issues/35#issuecomment-348636665 "2017-12-01T22:59:15Z - Replied by Github Reply Comments"):
> Could it be possible to also add a timer so it can either auto assign or assign after a selected time?  For instance, when the pug raiders join our server the Lead running the raid gives them @raiders, new members get a different rank.  I'd like the Lead to be able to assign the raiders and if after say 5 minutes if they are not assigned we auto assign.

So you'd want to have it check after a certain time if they've been assigned to a role? If they haven't, you'd then want it to auto-assign them to a specified role?

**Comment by netwrksolutions** [comment]
Yes.  Since everyone comes in as @everyone, I'd want to allow a certain time for Mods/Admins to assign them.  If they remain unassigned after a set time frame I can use the bot to dump them to @xNifty guest.

**Comment by xNifty** [comment]
[**@netwrksolutions**](https://github.com/netwrksolutions) commented on [Dec 2, 2017, 12:43 PM EST](https://github.com/xNifty/niftybot-discord/issues/35#issuecomment-348707658 "2017-12-02T17:43:40Z - Replied by Github Reply Comments"):
> Yes.  Since everyone comes in as @everyone, I'd want to allow a certain time for Mods/Admins to assign them.  If they remain unassigned after a set time frame I can use the bot to dump them to @guest.

That *should* be possible to do, but I'll have to play around with it a bit.

**Comment by xNifty** [comment]
Let's start by clarifying that I did not implement any form of timer.  This happens right on join or not at all.  It also requires some changes to the config file for the server, which by default isn't added to existing configs (although I added a new function (4ea023282a8f8f2132ba0c883d36d777df4a9313) which could be used to do this).
This feature was finished in 986bd72fc7e9dad151d3836fe1e824c5edfa8f48
